server/cmwell-bg/src/test/scala/cmwell/bg/test/BGResilienceSpec.scala

```diff
@@ -99,34 +95,32 @@ class BGResilienceSpec extends FlatSpec with BeforeAndAfterAll with BgEsCasKafka
     }
 
     // send them all
-    pRecords.foreach { kafkaProducer.send(_)}
-
-    // scalastyle:off
-    println("waiting for 20 seconds")
-    Thread.sleep(20000)
-    // scalastyle:on
+    pRecords.foreach {kafkaProducer.send(_)}
 
-    for( i <- 0 until numOfCommands) {
-      val nextResult = Await.result(irwService.readPathAsync(s"/cmt/cm/bg-test/circumvented_bg/info$i"), 5.seconds)
-      withClue(nextResult, s"/cmt/cm/bg-test/circumvented_bg/info$i"){
-        nextResult should not be empty
-      }
+    val casCheck = Future.traverse((0 until numOfCommands).toList) {i =>
+      cmwell.util.concurrent.spinCheck(1.second, true, 60.seconds) {
+        irwService.readPathAsync(s"/cmt/cm/bg-test/circumvented_bg/info$i")} ( _.nonEmpty )
     }
 
-    for( i <- 0 until numOfCommands) {
-      val searchResponse = Await.result(
+    val esCheck = cmwell.util.concurrent.spinCheck(1.second, true, 60.seconds) {
+      try{
         ftsServiceES.search(
-          pathFilter = None,
-          fieldsFilter = Some(SingleFieldFilter(Must, Equals, "system.path", Some(s"/cmt/cm/bg-test/circumvented_bg/info$i"))),
-          datesFilter = None,
-          paginationParams = PaginationParams(0, 200)
-        ),
-        10.seconds
+        pathFilter = None,
```

Comment:
You could use the pathFilter with descendants instead of manually doing it with system.parent.parent_hierarchy